search/2015-02-28-Preview/swagger/searchindex.json

```diff
@@ -0,0 +1,280 @@
+        }
+      },
+      "description": "Parameters for filtering, sorting, faceting, paging, and other search query behaviors.",
+      "x-ms-external": true
+    },
+    "SuggestParameters": {
+      "properties": {
+        "filter": {
+          "externalDocs": {
+            "url": "https://msdn.microsoft.com/library/azure/dn798921.aspx"
+          },
+          "type": "string",
+          "description": "Gets or sets the OData $filter expression to apply to the suggestions query."
+        },
+        "highlightPostTag": {
+          "type": "string",
+          "description": "Gets or sets a string tag that is appended to hit highlights. Must be set with HighlightPreTag. If omitted, hit highlighting of suggestions is disabled."
+        },
+        "highlightPreTag": {
+          "type": "string",
+          "description": "Gets or sets a string tag that is prepended to hit highlights. Must be set with HighlightPostTag. If omitted, hit highlighting of suggestions is disabled."
+        },
+        "minimumCoverage": {
+          "type": "number",
+          "format": "double",
+          "description": "Gets or sets a number between 0 and 100 indicating the percentage of the index that must be covered by a suggestion query in order for the query to be reported as a success. This parameter can be useful for ensuring search availability even for services with only one replica. The default is 80."
+        },
+        "orderBy": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Gets or sets the list of OData $orderby expressions by which to sort the results. Each expression can be either a field name or a call to the geo.distance() function. Each expression can be followed by asc to indicate ascending, and desc to indicate descending. The default is ascending order. Ties will be broken by the match scores of documents. If no OrderBy is specified, the default sort order is descending by document match score. There can be at most 32 Orderby clauses."
+        },
+        "searchFields": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Gets or sets the list of field names to consider when querying for suggestions."
+        },
+        "select": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Gets or sets the list of fields to retrieve. If unspecified, all fields marked as retrievable in the schema are included."
+        },
+        "top": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Gets or sets the number of suggestions to retrieve. This must be a value between 1 and 100. The default is to 5."
```

Comment:
Default values and all validation are done server-side. What does AutoRest do with this information? We specifically *do not* want default values sent in the request. Client-side validation would be ok, but I suspect the error messages from our API would be better.